### PR TITLE
Add PhysicalConstant namespace to use consistent values

### LINKED ISSUE #54: Add PhysicalConstant namespace to use consistent values throughout the code
## Reason
Using locally-defined physical constants can lead to inconsistency in what values are used for physical constants such as the ideal gas constant and such. Using non-consistent values can lead to discrepancy in predictions between different classes, and... It' just not good. 

## Design
Using a PhysicalConstant namespace allows us to define values to be used throughout the code with accuracy and consistency. 

## Impact
This impacts current classes using the ideal gas constants, but the main impact will be to be able to use consistent values for physical constants in TMAP8.

### PR DESCRIPTION
**Pull request by simopier**
- Create PhysicalConstants namespace
- Update existing classes to use PhysicalConstants.h
    
(Close #54)

### CODE REVIEW

**Comment by cticenhour** [comment]
Oops, I think you might have based this off main - lots of merge commits!

**Comment by simopier** [comment]
> Oops, I think you might have based this off main - lots of merge commits!

Yes, I was like: OMG what did I do again ?!?!? I'm working on fixing it to remove all these commits. Sorry!

**Comment by simopier** [comment]
That should be better @cticenhour. 😮‍💨

File: include/utils/PhysicalConstants.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+/*   Copyright 2021 - 2023 Battelle Energy Alliance, LLC    */
+/*                   ALL RIGHTS RESERVED                    */
+/************************************************************/
+
+#pragma once
+
+namespace PhysicalConstants
+{
+// Physical constants
+// Avogadro's number (at/mol)
```
Comment (cticenhour) [inline include/utils/PhysicalConstants.h:14]:
```suggestion
// Avogadro's number (atoms/mol)
```
Prefer to type `at` out for this one

**Review by cticenhour — CHANGES_REQUESTED** [review]
I like this idea! Did something similar for the EM module, and it helped keep things consistent. A few units typos I noticed.

**Review by cticenhour — APPROVED** [review]

**Pull request merged**